langchain4j-oracle/src/main/java/dev/langchain4j/store/embedding/oracle/OracleEmbeddingStore.java

```diff
@@ -134,12 +133,15 @@ private static void createIndex(Builder builder) throws SQLException {
             if (builder.vectorIndexCreateOption == CreateOption.CREATE_OR_REPLACE)
                 statement.addBatch("DROP INDEX IF EXISTS " + indexName);
 
+            // The COSINE metric used here should match the VECTOR_DISTANCE metric of the search method.
             statement.addBatch("CREATE VECTOR INDEX IF NOT EXISTS " + indexName +
                         " ON " + tableName + "(" + builder.embeddingTable.embeddingColumn() + ")" +
                         " ORGANIZATION NEIGHBOR PARTITIONS" +
-                        " WITH DISTANCE " + builder.distanceMetric.name());
+                        " WITH DISTANCE COSINE");
 
             statement.executeBatch();
+        } catch (SQLException sqlException) {
+            log.warn("Error creating index: " + sqlException.getMessage(), sqlException);
```

Comment:
```suggestion
            throw new RuntimeException(sqlException);
```
Shouldn't we propagate this exception to the user?

Reply:
Yes it should, the user explicitly asked for the index to be created, the exception should be propagated